Mathlib/Topology/Separation.lean

```diff
@@ -2286,51 +2299,91 @@ instance (priority := 100) NormalSpace.of_compactSpace_r1Space [CompactSpace X]
     NormalSpace X where
   normal _s _t hs ht := .of_isCompact_isCompact_isClosed hs.isCompact ht.isCompact ht
 
-/-- A regular topological space with second countable topology is a normal space.
+/-- Lemma used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+lemma countable_covers_witnessing_separated_nhds {h k : Set X}
```

Comment:
It is somewhat strange to have this two long assumptions `h_cov` and `k_cov` at the beginning: is it something that you would expect to use in other occasions or are there just extremely localised properties of a pair of covers? Because in the first case I would suggest that you create a `definition` or a `structure` to contain these properties, in the same spirit as `SperatedNhds`

Reply:
Hmm, this occurred to me at the time. I actually think `h_cov /\ k_cov <-> SeparatedNhds h k`, (oh of course, for the other direction, u is the constant function yielding the appropriate separating nhd). I'll play around with this, thanks.

Reply:
Introduced as `SeparatingCover` in [9daaadf](https://github.com/leanprover-community/mathlib4/pull/13176/commits/9daaadff9c8cd293824a72bb2cef14ad5459780f)